src/order/heyting_algebra.lean

```diff
@@ -0,0 +1,136 @@
+/-
+Copyright (c) 2020 Chris A. Upshaw. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris A. Upshaw.
+
+Type class hierarchy for Heyting algebras.
+-/
+import order.bounded_lattice
+
+set_option old_structure_cmd true
+
+/-!
+# Heyting algebra
+
+In this file we introduce Heyting algebras.
+
+A Heyting algebra is a bounded distributed lattice with an internal implication.
+
+Having an internal implication and a bottom allows for defining an (possilbe intuitionistic)
+complement.
+
+## Notation
+
+ - ` ⇨ ` : implication as an operator internal to a lattice.
+
```

Comment:
You should include a section on the main definitions and results; see the [documentation guidelines](https://leanprover-community.github.io/contribute/doc.html#header-comment).

Also, do you mind adding a section with a reference or two?